eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -18,15 +18,29 @@ jobs:
   variables:
     displayNameFilter: $[ coalesce(variables.jobMatrixFilter, '.*') ]
   pool:
-    name: Azure Pipelines
-    vmImage: ubuntu-18.04
+    name: azsdk-pool-mms-ubuntu-1804-general
+    vmImage: MMSUbuntu18.04
   displayName: Generate Job Matrix
   steps:
+    - checkout: none
+
+    - pwsh: |
+        git clone --no-checkout --filter=tree:0 git://github.com/$(Build.Repository.Name).git .
```

Comment:
I tried putting this in a separate script, but of course the pipeline can't load any external files until it does a checkout!